src/concerto/metamodel@0.3.0.cto

```diff
@@ -0,0 +1,191 @@
+  @FormEditor("hide", true)
+  o Decorator[] decorators optional
+  o Boolean isAbstract default=false
+  o Identified identified optional
+  @FormEditor("title", "Super Type")
+  o TypeIdentifier superType optional
+  o Property[] properties
+}
+
+concept AssetDeclaration extends ConceptDeclaration {
+}
+
+concept ParticipantDeclaration extends ConceptDeclaration {
+}
+
+concept TransactionDeclaration extends ConceptDeclaration {
+}
+
+concept EventDeclaration extends ConceptDeclaration {
+}
+
+@FormEditor("defaultSubconcept","concerto.metamodel.StringProperty")
+abstract concept Property {
+  o String name default="propertyName" regex=/^(?!null|true|false)(\p{Lu}|\p{Ll}|\p{Lt}|\p{Lm}|\p{Lo}|\p{Nl}|\$|_|\\u[0-9A-Fa-f]{4})(?:\p{Lu}|\p{Ll}|\p{Lt}|\p{Lm}|\p{Lo}|\p{Nl}|\$|_|\\u[0-9A-Fa-f]{4}|\p{Mn}|\p{Mc}|\p{Nd}|\p{Pc}|\u200C|\u200D)*$/u
+  @FormEditor("title", "Is Array?")
+  o Boolean isArray default=false
+  @FormEditor("title", "Is Optional?")
+  o Boolean isOptional default=false
+  @FormEditor("hide", true)
+  o Decorator[] decorators optional
+}
+
+concept RelationshipProperty extends Property {
+  @FormEditor("title", "Type Name", "selectOptions", "types")
+  o TypeIdentifier type
+}
+
+concept ObjectProperty extends Property {
+  @FormEditor("hide", true)
+  o String defaultValue optional
+  @FormEditor("title", "Type Name", "selectOptions", "types")
+  o TypeIdentifier type
+}
+
+concept BooleanProperty extends Property {
+  @FormEditor("hide", true)
+  o Boolean defaultValue optional
+}
+
+concept DateTimeProperty extends Property {
+}
+
+concept StringProperty extends Property {
+  @FormEditor("hide", true)
+  o String defaultValue optional
+  @FormEditor("hide", true)
+  o StringRegexValidator validator optional
+}
+
+concept StringRegexValidator {
+  o String regex
+}
+
+concept DoubleProperty extends Property {
+  o Double defaultValue optional
+  o DoubleDomainValidator validator optional
+}
+
+concept DoubleDomainValidator {
+  o Double lower optional
+  o Double upper optional
+}
+
+concept IntegerProperty extends Property {
+  @FormEditor("hide", true)
+  o Integer defaultValue optional
+  @FormEditor("hide", true)
+  o IntegerDomainValidator validator optional
+}
+
+concept IntegerDomainValidator {
+  o Integer lower optional
+  o Integer upper optional
+}
+
+concept LongProperty extends Property {
+  @FormEditor("hide", true)
+  o Long defaultValue optional
+  @FormEditor("hide", true)
+  o LongDomainValidator validator optional
+}
+
+concept LongDomainValidator {
+  o Long lower optional
+  o Long upper optional
+}
+
+abstract concept Import {
+  o String namespace
+  o String uri optional
+}
+
+concept ImportAll extends Import {
+}
+
+concept ImportType extends Import {
+  o String name
+}
+
+concept ModelFile {
```

Comment:
Should this be `concept ModelFile identified by namespace {`

Reply:
🤔 interesting one! That makes sense to me.

Reply:
Side question: should we have a version in there...

Reply:
Not changed, as discussed.